pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -528,7 +528,7 @@ public void execute(Runnable task, CompletableFuture<?> future) {
         try {
             executor.execute(task);
         } catch (Throwable t) {
-            future.completeExceptionally(t);
+            executor.execute(() -> future.completeExceptionally(t));
```

Comment:
Based on the Javadoc, we only expect to catch this when the executor is shutting down. I don't think we want to submit this task for all cases.

Reply:
This change has been removed.

Reply:
>> Now, if we call `put /a`, it will use the `event thread` of the ZK client after the operation; but if we call `put /a/b/c`(the node `/a` is not exists), it will use the `metadata store thread` after the operation.

> What is the consequence of using different threads? Did you see out of order events?
> EDIT: looks like the consequence is this error `ZooKeeper session reconnection timeout. Notifying session is lost`.

To easier to trace the context, I answer this question here.

No, I notice this when I check for deadlocks https://github.com/apache/pulsar/pull/20189. I saw that the ZK thread was executing the task of creating the topic. And then I realized that maybe @merlimat  forgot to change this.